### PR TITLE
Docs reorganization!

### PR DESCRIPTION
**Pull request by abehaskins**
**Don't merge yet!**

Creating a PR for comments. Please add your thoughts.

### CODE REVIEW

**Comment by abehaskins** [comment]
My biggest problem with these changes is the "time to first snippet" is increased in exchange for organization and readability. I think perhaps the root README could include screenshots / simple snippet just for marketing purposes.

**Comment by samtstern** [comment]
This change LGTM.  I am not worried about the time to first snippet change.

Only concern is a lack of description for the features on the front page.  Not sure why anyone would click on 'Database' in the table of contents since the concept of Database UI is not so obvious.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,62 @@
...
+    [individual CLA](https://developers.google.com/open-source/cla/individual).
+  * If you work for a company that wants to allow you to contribute your work,
+    then you'll need to sign a
+    [corporate CLA](https://developers.google.com/open-source/cla/corporate).
+
+Follow either of the two links above to access the appropriate CLA and
+instructions for how to sign and return it. Once we receive it, we'll be able to
+accept your pull requests.
+
+### Contribution Process
```
Comment (mcdonamp) [inline CONTRIBUTING.md]: Seems like this shouldn't be under "Legal Process"

File: README.md [hunk 2]
```diff
@@ -10,126 +10,16 @@ A compatible FirebaseUI client is also available for [iOS](https://github.com/fi
 
 ## Table of Content
```
Comment (mcdonamp) [inline README.md]: "Contents"

File: auth/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # FirebaseUI for Android — Auth
```
Comment (alfongj) [inline auth/README.md]: I think we need to make the value proposition more compelling on this initial page.

I would propose to have here something like [LeakCanary's description page](https://github.com/square/leakcanary):
1) How does this help me? -> Big GIF or some other representation of what you get with FirebaseUI Auth
2) How much will it cost me? -> Let's put a snippet with the minimum code needed to run it. I'd say: gradle includes + a snippet with the check of whether the user is signed-in and if not, an invocation to the sign-in flow with Google and Email, that don't require any extra configuration.
3) How can I learn more? -> Pointers to advanced configuration and so on.

**Comment by samtstern** [comment]
@abehaskins what's the status of this PR?

**Comment by SUPERCILEX** [comment]
@abehaskins Any updates on this?

**Comment by samtstern** [comment]
@abehaskins there's nearly no way you can resolve these merge conflicts in any sane way, things have moved a lot since you opened this.  I am gonna close this for repo health reasons, but I would still love to reorganize the docs if you want to.

**Pull request closed without merging**